merkle/crdt/merklecrdt.go

```diff
@@ -65,19 +61,28 @@ func (base *baseMerkleCRDT) Value(ctx context.Context) ([]byte, error) {
 	return base.crdt.Value(ctx)
 }
 
-func (base *baseMerkleCRDT) ID() string {
-	return base.crdt.ID()
-}
-
-// Publishes the delta to state.
-func (base *baseMerkleCRDT) Publish(
```

Comment:
suggestion: I am hesitant to remove this Publish function as it keeps a cleaner seperation of concerns when writting MerkleCRDT delta-mutator functions.

The recipe is simple at the moment. Call the core crdt equivalent mutator, then call Publish(delta). 

Even if Publish is just calling `clock.AddDAGNode` under the hood, it feels like its needlessly bleeding abstractions, and lose on some of the purity from before.

Reply:
The way I saw it is that when we called `merkleCRDT.Set(...)` and there was a call to `Publish`, we had to go to `Publish` only to find out that it was only calling `clock.AddDAGNode` and I found it obstructive more than anything. The way it was been changed makes the flow very clear. We set the CRDT delta object and then send that object to the clock for insertion into the DAG. Can't get purer than that if you ask me.

Reply:
Just noting that I agree with Fred here - I find the new, less-abstract flow easier to read.